### PR TITLE
*: make 'grant all privileges' work right

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

See https://github.com/pingcap/tidb/pull/11449

### What is changed and how it works?

Add those privileges

```
	CreateTMPTablePriv: "Create_tmp_table_priv",
	LockTablesPriv:     "Lock_tables_priv",
	CreateRoutinePriv:  "Create_routine_priv",
	AlterRoutinePriv:   "Alter_routine_priv",
	EventPriv:          "Event_priv",
```

Make Priv2UserCol, AllGlobalPrivs consistent and many other changes.

### Check List 

Tests 

 - Unit test

Side effects

 - Breaking backward compatibility

### CODE REVIEW

File: mysql/const_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+func (s *testConstSuite) TestPrivAllConsistency(c *C) {
+	// AllPriv in mysql.user columns.
+	for priv := PrivilegeType(CreatePriv); priv != AllPriv; priv = priv << 1 {
+		_, ok := Priv2UserCol[priv]
+		c.Assert(ok, IsTrue, Commentf("priv fail %d", priv))
+	}
+
+	for _, v := range AllGlobalPrivs {
+		_, ok := Priv2UserCol[v]
+		c.Assert(ok, IsTrue)
```
Comment (kennytm) [inline mysql/const_test.go]:
```go
c.Assert(Priv2UserCol, HasKey, v)
```

Comment (imtbkcat) [reply mysql/const_test.go]: please address comment @tiancaiamao

**Comment by imtbkcat** [comment]
ping @tiancaiamao

**Comment by imtbkcat** [comment]
@tiancaiamao please cherry-pick this pr to `release-2.1` branch.

**Pull request merged**